kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/http/AbstractHttpClientNewWebSocketBuilderTest.java

```diff
@@ -93,32 +93,55 @@ void buildAsyncCantUpgradeThrowsWebSocketHandshakeException() {
   }
 
   @Test
-  void buildAsyncIncludesRequiredHeadersAndPropagatesConfigured() throws Exception {
-    server.expect().withPath("/websocket-headers-test").andReturn(200, "hello").once();
-    final AtomicBoolean open = new AtomicBoolean(false);
-    try {
-      httpClient.newWebSocketBuilder()
-          .header("A-Random-Header", "A-Random-Value")
-          .subprotocol("amqp")
-          .uri(URI.create(server.url("/websocket-headers-test")))
-          .buildAsync(new WebSocket.Listener() {
-            @Override
-            public void onOpen(WebSocket webSocket) {
-              open.set(true);
-            }
-          }).get(10L, TimeUnit.SECONDS);
-    } catch (ExecutionException e) {
-      WebSocketHandshakeException wshe = (WebSocketHandshakeException) e.getCause();
-      assertThat(wshe.getResponse().code()).isEqualTo(200);
```

Comment:
@manusa you should preserve this aspect of the test as well (as a separate case if you want).  To ensure that the handshake exception presents the underlying code - I couldn't find that we were testing that across the clients anywhere else.

Reply:
:+1: I'll add a new test-case for this

Reply:
there we go, there's an additional test with the specific case